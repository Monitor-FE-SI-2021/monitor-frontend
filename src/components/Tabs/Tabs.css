*, ::before, ::after {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
  }
  
  body {
    background: #fff;
  }
  
  .container {
    display: flex;
    flex-direction: column;
    position: relative;
    width: 100%;
    height: 100%;
    background: #f1f1f1;
  
    word-break: break-all;
    border: 1px solid rgba(0, 0, 0, 0.274);
  }
  
  .bloc-tabs {
    display: flex;
  }
  .tabs {
  
    padding: 15px;
    text-align: center;
    width: 50%;
    background: rgba(128, 128, 128, 0.075);
    cursor: pointer;
    border-bottom: 1px solid rgba(0, 0, 0, 0.274);
    box-sizing: content-box;
    position: relative;
    outline: none;
  }
  .tabs:not(:last-child){
    border-right: 1px solid rgba(0, 0, 0, 0.274);
  }
  
  .active-tabs  {
    background: white;
    border-bottom: 1px solid transparent;
  }
  
  button {
    border: none;
    
  }
  .content-tabs {
    flex-grow : 1;
  }
  .content {
    background: white;
    padding: 20px;
    width: 100%;
    height: 100%;
    display: none;
  }
  .content h2 {
    padding: 0px 0 5px 0px;
  }
  .content hr {
    width: 100px;
    height: 2px;
    background: #222;
    margin-bottom: 5px;
  }
  .content p {
    width: 100%;
    height: 100%;
  }
  .active-content {
    display: block;
  }

  .paragraph1 {
    color: black;
    background-color: gray;
    padding: 12px;
}

button {
    width: fit-content;
    padding: 10px 25px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    margin: 4px 2px;
    cursor: pointer;
    font-weight: bold;
}

.screenshot {
    /*background-color: #e3dddc;*/
    padding-bottom: 10px;
    max-width: 80%;
    margin-bottom: 20px;
}

.screenshot-img {
    width: 100%;
}
  