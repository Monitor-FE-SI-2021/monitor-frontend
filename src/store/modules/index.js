import { combineReducers } from 'redux'
import { connectRouter } from 'connected-react-router'
import menu from "./menu/menu";
import devices from "./devices/devices";
import groups from "./groups/groups";
import login from "./login/login";

// Kada napravite reducer (store module), dodajte ga ovdje ispod (counter, menu, noviModul, ...)

const createRootReducer = (history) => combineReducers({
    router: connectRouter(history),
    menu,
    devices,
    groups,
    login
})
export default createRootReducer