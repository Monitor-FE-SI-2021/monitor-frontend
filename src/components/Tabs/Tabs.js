import { useState } from "react";
import { useMemo } from "react";
import "./Tabs.css";
import { connect } from "react-redux";
import request, { wsEndpoint } from "../../service";
import { useDropzone } from "react-dropzone";
import React, { useCallback } from "react";
import Swal from "sweetalert2";
import { Route, Switch, useParams } from "react-router";
import FileManagerTable from "../FileManagerTable/FileManagerTable";
import Terminal from "../Terminal/Terminal";

const Tabs = (props, machine) => {
  let { name, tab } = useParams();

  const toggleTab = (tab) => {
    props.history.push(
      "/remotecontrol/" + (name == undefined ? "0" : name) + "/" + tab
    );
  };

  const [url, setUrl] = useState("slkadhjaksl");

  const handleClick = async () => {
    try {
      var odgovor = await request(wsEndpoint + "/screenshot", "post", machine);

      setUrl(odgovor.data.message);
    } catch (err) {}
  };

  const baseStyle = {
    flex: 1,
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    textAlign: "center",
    padding: "80px",
    borderWidth: 2,
    borderRadius: 2,
    borderColor: "#eeeeee",
    borderStyle: "dashed",
    backgroundColor: "#fafafa",
    color: "#bdbdbd",
    outline: "none",
    transition: "border .24s ease-in-out",
  };

  const activeStyle = {
    borderColor: "#2196f3",
  };

  const acceptStyle = {
    borderColor: "#00e676",
  };

  const rejectStyle = {
    borderColor: "#ff1744",
  };

  const onDrop = useCallback((acceptedFiles) => {
    //SEND FILES TO SERVER FROM HERE!
    Swal.fire({
      title: "File manager",
      text: "You have successfully uploaded your file!",
      type: "success",
    });
  }, []);

  const {
    getRootProps,
    getInputProps,
    isDragActive,
    isDragAccept,
    isDragReject,
  } = useDropzone({ onDrop });

  const style = useMemo(
    () => ({
      ...baseStyle,
      ...(isDragActive ? activeStyle : {}),
      ...(isDragAccept ? acceptStyle : {}),
      ...(isDragReject ? rejectStyle : {}),
    }),
    [isDragActive, isDragReject, isDragAccept]
  );

  return (
    <div className="container">
      <div className="bloc-tabs">
        <button
          className={
            tab == undefined || tab === "terminal" ? "tabs active-tabs" : "tabs"
          }
          onClick={() => toggleTab("terminal")}
        >
          Terminal
        </button>
        <button
          className={tab === "screenshot" ? "tabs active-tabs" : "tabs"}
          onClick={() => toggleTab("screenshot")}
        >
          Screenshot
        </button>
        <button
          className={tab === "filemanager" ? "tabs active-tabs" : "tabs"}
          onClick={() => toggleTab("filemanager")}
        >
          File Manager
        </button>
      </div>
      <div className="content-tabs">
        <div
          className={
            tab == undefined || tab === "terminal"
              ? "content  active-content"
              : "content"
          }
        >
          <div>
            <p className="paragraph1">Quick example</p>
            <div>
              <div className="screenshot">
                <p>Terminal</p>

                <Terminal machine={machine} />
              </div>
            </div>
          </div>
        </div>
        <div
          className={
            tab === "screenshot" ? "content  active-content" : "content"
          }
        >
          <div>
            <p className="paragraph1">Quick example</p>
            <div>
              <div className="screenshot">
                <p>Screenshot</p>
                <img
                  alt="Asked image will appear here."
                  className="screenshot-img"
                  src={`data:image/jpeg;base64,${url}`}
                />
              </div>
            </div>
          </div>

          <button type="button" onClick={handleClick}>
            Get Screenshot
          </button>
        </div>
        <div
          className={
            tab === "filemanager" ? "content  active-content" : "content"
          }
        >
          <FileManagerTable></FileManagerTable>
          <div className="uploadArea">
            <div {...getRootProps({ style })}>
              <input {...getInputProps()} />
              <p className="uploadText">
                Drag 'n' drop some files here, or click to select files
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default connect((state) => ({}), {})(Tabs);
