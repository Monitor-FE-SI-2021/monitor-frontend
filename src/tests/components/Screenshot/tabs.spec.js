import { Store } from '@material-ui/icons';
import React from 'react';
import renderer from 'react-test-renderer';
import {Tabs} from '../../../components/Tabs/Tabs'
import { Screenshot } from '../../../components/Tabs/Tabs'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store';
import store from '../../../store/store';


test('GetScreenshot test', () => {
    const component = renderer.create(
        <Provider store={store}>
         <Screenshot/> 
        </Provider>
        ,
    );
    let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
console.log(tree.children[1].props)
  tree.children[1].props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot()
});