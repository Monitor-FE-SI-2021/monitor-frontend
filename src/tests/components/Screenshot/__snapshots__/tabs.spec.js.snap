// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetScreenshot test 1`] = `
<div
  className="container tabs-container"
>
  <div
    className="content-tabs"
  >
    <div
      className="screenshot"
    >
      <p>
        Screenshot
      </p>
      <img
        alt="Asked image will appear here."
        className="screenshot-img"
        src="data:image/jpeg;base64,slkadhjaksl"
      />
    </div>
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Get Screenshot
  </button>
</div>
`;

exports[`GetScreenshot test 2`] = `
<div
  className="container tabs-container"
>
  <div
    className="content-tabs"
  >
    <div
      className="screenshot"
    >
      <p>
        Screenshot
      </p>
      <img
        alt="Asked image will appear here."
        className="screenshot-img"
        src="data:image/jpeg;base64,slkadhjaksl"
      />
    </div>
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Get Screenshot
  </button>
</div>
`;
